libraries/botbuilder-azure/botbuilder/azure/blob_storage.py

```diff
@@ -81,39 +153,54 @@ async def write(self, changes: Dict[str, object]):
             e_tag = None if e_tag == "*" else e_tag
             if e_tag == "":
                 raise Exception("blob_storage.write(): etag missing")
+
             item_str = self._store_item_to_str(item)
-            try:
-                self.client.create_blob_from_text(
-                    container_name=self.settings.container_name,
-                    blob_name=name,
-                    text=item_str,
-                    if_match=e_tag,
+
+            if e_tag:
+                await blob_reference.upload_blob(
+                    item_str, match_condition=MatchConditions.IfNotModified, etag=e_tag
                 )
-            except Exception as error:
-                raise error
+            else:
+                await blob_reference.upload_blob(item_str, overwrite=True)
```

Comment:
Match conditions and eTags changed quite a bit and I'm not 100% confident that I implemented this correctly. I tried some other approaches and this was the one that made all of our tests pass. Please review this part closely.